src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IAnonymousFunctionExpression.cs

```diff
@@ -338,14 +342,16 @@ static void F()
       IVariableInitializer (OperationKind.VariableInitializer, IsInvalid) (Syntax: '= (Action<i ... (() => F())')
         IDelegateCreationExpression (OperationKind.DelegateCreationExpression, Type: System.Action<System.Int32>, IsInvalid) (Syntax: '(Action<int>)(() => F())')
           Target: 
-            IAnonymousFunctionExpression (Symbol: lambda expression) (OperationKind.AnonymousFunctionExpression, Type: null, IsInvalid) (Syntax: '() => F()')
-              IBlockStatement (1 statements) (OperationKind.BlockStatement, IsInvalid, IsImplicit) (Syntax: 'F()')
-                IExpressionStatement (OperationKind.ExpressionStatement, IsInvalid, IsImplicit) (Syntax: 'F()')
-                  Expression: 
-                    IInvocationExpression (void Program.F()) (OperationKind.InvocationExpression, Type: System.Void, IsInvalid) (Syntax: 'F()')
-                      Instance Receiver: 
-                        null
-                      Arguments(0)
+            IParenthesizedExpression (OperationKind.ParenthesizedExpression, Type: null, IsInvalid) (Syntax: '(() => F())')
```

Comment:
>IParenthesizedExpression (OperationKind.ParenthesizedExpression, Type: null, IsInvalid) (Syntax: '(() => F())') [](start = 12, length = 111)

Do we have tests for scenario like ```CType((Sub() System.Console.WriteLine()), System.Action)``` for VB? Please add one. Similar for implicit conversion

Reply:
It looks like we are missing an implicit conversions case for C#. 

---
In reply to: [147013844](https://github.com/dotnet/roslyn/pull/22846#discussion_r147013844) [](ancestors = 147013844)

Reply:
Should also test erroneous conversions, i.e. when IConversion  node is used instead of IDelegateCreationExpression

---
In reply to: [147015069](https://github.com/dotnet/roslyn/pull/22846#discussion_r147015069) [](ancestors = 147015069,147013844)

Reply:
This caused an assert to fire: [https://github.com/dotnet/roslyn/issues/22879](https://github.com/dotnet/roslyn/issues/22879)

---
In reply to: [147013844](https://github.com/dotnet/roslyn/pull/22846#discussion_r147013844) [](ancestors = 147013844)